File: Cc.Core/Models/Product.cs
using System;
using System.Collections.Generic;

namespace Cc.Core.Models;

public partial class Product
{
    public long Id { get; set; }

    public string Description { get; set; } = null!;

    public double Price { get; set; }

    public string Code { get; set; } = null!;

    public long CompanyId { get; set; }

    public DateTime InsertDate { get; set; }

    public string? InsertUser { get; set; }

    public DateTime UpdateDate { get; set; }

    public string? UpdateUser { get; set; }

    public long? CategoryId { get; set; }

    public virtual ICollection<BroadcastingProduct> BroadcastingProduct { get; set; } = new List<BroadcastingProduct>();

    public virtual Category? Category { get; set; }

    public virtual Company Company { get; set; } = null!;

    public virtual ICollection<Customer> Customer { get; set; } = new List<Customer>();

    public virtual ICollection<CustomerRequiredField> CustomerRequiredField { get; set; } = new List<CustomerRequiredField>();

    public virtual ICollection<Investment> Investment { get; set; } = new List<Investment>();

    public virtual ICollection<OrderRow> OrderRow { get; set; } = new List<OrderRow>();

    public virtual ICollection<PrincipalAuth> PrincipalAuth { get; set; } = new List<PrincipalAuth>();

    public virtual ICollection<WarehouseMovement> WarehouseMovement { get; set; } = new List<WarehouseMovement>();
}

File: Cc.Ui/src/models/product.ts
import { Entity, resource, field, required, association } from '../framework/entity';
import { Pagination, Pager } from '../framework/core/actions';
import { Company } from './company';
import { cache } from '../app/cache';
import { LookupKey } from '../core/lookup';
import { Category } from './category';

@resource('product')
export class Product extends Entity {

    @field()
    id: number;
    @field()
    description: string;
    @field()
    price: number;
    @field()
    code: string;
    @field()
    companyId: number;
    @field()
    insertDate: string;
    @field()
    insertUser: string;
    @field()
    updateDate: string;
    @field()
    updateUser: string;
    @field()
    categoryId: number;

    @association(Company, 'companyId', (id: string | number) => cache(LookupKey.COMPANY, id))
    company: Company;
    @association(Category, 'categoryId', (id: string | number) => cache(LookupKey.CATEGORY, id))
    category: Category;
    
    constructor(data?: any) {
        super()
        this.init(data);      
    }

    static newItem(): Product {

        let item = new Product();
        return item;
    }
}

@resource('product')
export class ProductSearchModel extends Entity implements Pagination {
    
    @field()
    description?: string;
    @field()
    price?: number;
    @field()
    code?: string;
    @field()
    companyId?: number;
    @field()
    categoryId?: number;
    @field()
    disabled?: boolean;

    @field()
    pager: Pager = <Pager> {
        take: 100,
        orderBy: 'id'
    };
}